### PR TITLE
OEL-76: Drupal 9 compatibility fixes.

### PR DESCRIPTION
**Pull request by saidatom**

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -23,14 +23,16 @@ services:
     environment:
       - MYSQL_ALLOW_EMPTY_PASSWORD=yes
   selenium:
+    # Newer Selenium Chrome images are not supported at the moment by behat/mink-selenium2-driver.
```
Comment (yenyasinn) [inline .drone.yml]:
```suggestion
```
We use this image in all components, so no need to add comment.

File: docker-compose.yml [hunk 2]
```diff
@@ -37,7 +39,9 @@ services:
       - VNC_NO_PASSWORD=1
     ports:
       - '4444:4444'
-      - '5900:5900'
+      - "5900:5900"
```
Comment (yenyasinn) [inline docker-compose.yml]:
```suggestion
      - '5900:5900'
```
single comma looks consistent with previous port declaration

File: docker-compose.yml [hunk 3]
```diff
@@ -37,7 +39,9 @@ services:
...
+    volumes:
```
Comment (yenyasinn) [inline docker-compose.yml]:
```suggestion
```
Do we really need to define `volumes`?

File: modules/entity_version_history/entity_version_history.info.yml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 name: Entity version history
```
Comment (yenyasinn) [inline modules/entity_version_history/entity_version_history.info.yml:1]:
```suggestion
name: 'Entity version history'
```

File: modules/entity_version_history/entity_version_history.info.yml [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
 description: Entity version history provides a tab where version history of the entity is visualised in a table format.
```
Comment (yenyasinn) [inline modules/entity_version_history/entity_version_history.info.yml:2]:
```suggestion
description: 'Entity version history provides a tab where version history of the entity is visualised in a table format.'
```

File: modules/entity_version_workflows/entity_version_workflows.info.yml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 name: Entity version workflows
```
Comment (yenyasinn) [inline modules/entity_version_workflows/entity_version_workflows.info.yml:1]:
```suggestion
name: 'Entity version workflows'
```

File: modules/entity_version_workflows/entity_version_workflows.info.yml [hunk 7]
```diff
@@ -1,7 +1,7 @@
...
 description: Entity version workflow allows to configure and control version numbers through workflow state transitions.
```
Comment (yenyasinn) [inline modules/entity_version_workflows/entity_version_workflows.info.yml:2]:
```suggestion
description: 'Entity version workflow allows to configure and control version numbers through workflow state transitions.'
```

File: modules/entity_version_workflows/modules/entity_version_workflows_example/entity_version_workflows_example.info.yml [hunk 8]
```diff
@@ -1,7 +1,7 @@
 name: Entity Version Workflows Example
```
Comment (yenyasinn) [inline modules/entity_version_workflows/modules/entity_version_workflows_example/entity_version_workflows_example.info.yml:1]:
```suggestion
name: 'Entity Version Workflows Example'
```

File: modules/entity_version_workflows/modules/entity_version_workflows_example/entity_version_workflows_example.info.yml [hunk 9]
```diff
@@ -1,7 +1,7 @@
...
 description: Example module to test the entity version content workflow functionalities.
```
Comment (yenyasinn) [inline modules/entity_version_workflows/modules/entity_version_workflows_example/entity_version_workflows_example.info.yml:2]:
```suggestion
description: 'Example module to test the entity version content workflow functionalities.'
```

File: tests/modules/entity_version_test/entity_version_test.info.yml [hunk 10]
```diff
@@ -1,7 +1,7 @@
 name: Entity Version Test
```
Comment (yenyasinn) [inline tests/modules/entity_version_test/entity_version_test.info.yml:1]:
```suggestion
name: 'Entity Version Test'
```

File: tests/modules/entity_version_test/entity_version_test.info.yml [hunk 11]
```diff
@@ -1,7 +1,7 @@
...
 description: Test module to test the entity version functionalities.
```
Comment (yenyasinn) [inline tests/modules/entity_version_test/entity_version_test.info.yml:2]:
```suggestion
description: 'Test module to test the entity version functionalities.'
```

File: modules/entity_version_history/entity_version_history.info.yml (hunk 4 above)
Comment (claudiu-cristea) [reply modules/entity_version_history/entity_version_history.info.yml:1]: Even without quotes it's a valid YAML syntax. Not to say that this is very, very out of scope.

File: modules/entity_version_history/entity_version_history.info.yml (hunk 5 above)
Comment (claudiu-cristea) [reply modules/entity_version_history/entity_version_history.info.yml:2]: Same here. Severely out of scope.

File: modules/entity_version_history/entity_version_history.info.yml (hunk 4 above)
Comment (yenyasinn) [reply modules/entity_version_history/entity_version_history.info.yml:1]: Drupal wraps module name and description by commas. We update it in other components during D9 compatibility fixes.

Comment (claudiu-cristea) [reply modules/entity_version_history/entity_version_history.info.yml:1]: Well, if Drupal does it, OK. Is there a Drupal coding standard in this regard?

Comment (yenyasinn) [reply modules/entity_version_history/entity_version_history.info.yml:1]: There isn't standard about it. But actually it wasn't decided how it should be in Drupal core. Currently it is inconsistent there also. Please have a look https://www.drupal.org/project/drupal/issues/2994928.

File: modules/entity_version_workflows/src/EntityVersionWorkflowManager.php [hunk 12]
```diff
@@ -91,7 +91,10 @@ public function updateEntityVersion(ContentEntityInterface $entity, $field_name)
     // Compute the transition being used in order to get the version actions
     // from its config. For this, we need to load the latest revision of the
     // entity.
-    $revision = $this->moderationInfo->getLatestRevision($entity->getEntityTypeId(), $entity->id());
+    /* @var \Drupal\Core\Entity\RevisionableStorageInterface $entity_storage */
```
Comment (nagyad) [inline modules/entity_version_workflows/src/EntityVersionWorkflowManager.php]: Below you have $storage not $entity_storage

File: modules/entity_version_workflows/src/EntityVersionWorkflowManager.php [hunk 13]
```diff
@@ -153,7 +156,11 @@ protected function isEntityChanged(ContentEntityInterface $entity): bool {
     $field_blacklist = $event->getFieldBlacklist();
 
     // We consider the latest revision as original to compare with the entity.
-    $latestRevision = $this->moderationInfo->getLatestRevision($entity->getEntityTypeId(), $entity->id());
+    /* @var \Drupal\Core\Entity\RevisionableStorageInterface $entity_storage */
```
Comment (nagyad) [inline modules/entity_version_workflows/src/EntityVersionWorkflowManager.php]: Below you have $storage not $entity_storage

**Review by nagyad — CHANGES_REQUESTED** [review]

**Review by nagyad — DISMISSED** [review]

File: composer.json [hunk 14]
```diff
@@ -6,26 +6,25 @@
...
-        "drush/drush": "~9.0@stable",
-        "guzzlehttp/guzzle": "~6.3",
-        "openeuropa/code-review": "~1.5",
-        "openeuropa/drupal-core-require-dev": "^8.7",
-        "openeuropa/task-runner": "~1.0.0-beta5",
-        "phpunit/phpunit": "~6.0",
-        "symfony/dom-crawler": "~3.4"
+        "drupal/drupal-extension": "~4.0",
+        "drush/drush": "^10.3",
+        "openeuropa/behat-transformation-context": "~0.1",
```
Comment (sergepavle) [inline composer.json]: Seems like we don't need to add this component.
In this component, we don't use behat tests.

File: composer.json [hunk 15]
```diff
@@ -6,26 +6,25 @@
...
+        "drupal/core-composer-scaffold": "^8.9 || ^9.1",
         "drupal/config_devel": "~1.2",
-        "drush/drush": "~9.0@stable",
-        "guzzlehttp/guzzle": "~6.3",
-        "openeuropa/code-review": "~1.5",
-        "openeuropa/drupal-core-require-dev": "^8.7",
-        "openeuropa/task-runner": "~1.0.0-beta5",
-        "phpunit/phpunit": "~6.0",
-        "symfony/dom-crawler": "~3.4"
+        "drupal/drupal-extension": "~4.0",
```
Comment (sergepavle) [inline composer.json]: Seems like we don't need to add this component.
In this component, we don't use behat tests.

**Review by sergepavle — COMMENTED** [review]
Looks good and works well.
I've left some very minor comments.

**Review by sergepavle — DISMISSED** [review]

**Review by nagyad — DISMISSED** [review]

File: composer.json [hunk 16]
```diff
@@ -6,26 +6,23 @@
...
         "drupal/config_devel": "~1.2",
-        "drush/drush": "~9.0@stable",
-        "guzzlehttp/guzzle": "~6.3",
-        "openeuropa/code-review": "~1.5",
-        "openeuropa/drupal-core-require-dev": "^8.7",
-        "openeuropa/task-runner": "~1.0.0-beta5",
-        "phpunit/phpunit": "~6.0",
-        "symfony/dom-crawler": "~3.4"
+        "drush/drush": "^10.3",
+        "openeuropa/code-review": "~1.6",
```
Comment (upchuk) [inline composer.json]: Let's use ^

File: modules/entity_version_workflows/src/Event/CheckEntityChangedEvent.php [hunk 17]
```diff
@@ -4,6 +4,12 @@
 
 namespace Drupal\entity_version_workflows\Event;
 
+/**
+ * Use Component instead of Contracts.
```
Comment (upchuk) [inline modules/entity_version_workflows/src/Event/CheckEntityChangedEvent.php]: No need for this comment. We use this everywhere, we cannot keep adding this comment everywhere where we throw events.

File: modules/entity_version_workflows/tests/src/Kernel/EntityVersionWorkflowsTest.php [hunk 18]
```diff
@@ -182,18 +175,19 @@ public function testEntityVersionNoUpdate() {
     $values = [
       'title' => 'Workflow node',
       'type' => 'entity_version_workflows_example',
-      'moderation_state' => 'draft',
     ];
     /** @var \Drupal\node\NodeInterface $node */
     $node = $this->nodeStorage->create($values);
+    $node->setNewRevision();
```
Comment (upchuk) [inline modules/entity_version_workflows/tests/src/Kernel/EntityVersionWorkflowsTest.php]: This is a change in the test. Why is this needed?

File: modules/entity_version_workflows/tests/src/Kernel/EntityVersionWorkflowsTest.php [hunk 19]
```diff
@@ -182,18 +175,19 @@ public function testEntityVersionNoUpdate() {
...
     $node->save();
 
     // There is no default value so all versions should be 0.
-    $this->assertEqual('draft', $node->moderation_state->value);
+    $this->assertEquals('draft', $node->moderation_state->value);
     $this->assertEntityVersion($node, 0, 0, 0);
 
     // Alter the node to increase the patch number (stay in draft).
     $node->set('title', 'New title');
+    $node->setNewRevision();
```
Comment (upchuk) [inline modules/entity_version_workflows/tests/src/Kernel/EntityVersionWorkflowsTest.php]: Same here. Why is this needed?

File: modules/entity_version_workflows/tests/Functional/EntityVersionNodeRevisionTest.php [hunk 20]
```diff
@@ -35,6 +35,11 @@ class EntityVersionNodeRevisionTest extends BrowserTestBase {
     'entity_version_workflows_example',
```
Comment (upchuk) [inline modules/entity_version_workflows/tests/Functional/EntityVersionNodeRevisionTest.php]: This test contains a usage of `drupalPostForm` which is deprecated and we need to change it.

**Review by upchuk — CHANGES_REQUESTED** [review]

**Review by upchuk — APPROVED** [review]

**Review by brummbar — APPROVED** [review]
Blind

File: .gitignore [hunk 21]
```diff
@@ -12,3 +12,4 @@ phpunit.xml
 runner.yml
 vendor
 yarn.lock
+/.gitattributes
```
Comment (claudiu-cristea) [inline .gitignore:15]: Requires a new empty line at the end.

File: tests/src/FunctionalJavascript/EntityVersionSettingsTest.php [hunk 22]
```diff
@@ -229,7 +233,7 @@ protected function assertFieldSelectOptions(string $name, array $expected_option
     });
     sort($options);
     sort($expected_options);
-    $this->assertIdentical($options, $expected_options);
+    $this->assertSame($options, $expected_options);
```
Comment (claudiu-cristea) [inline tests/src/FunctionalJavascript/EntityVersionSettingsTest.php:236]: As we've changed this line, let's swap the parameters. In PHPUnit the expectation comes first.

File: tests/src/FunctionalJavascript/EntityVersionSettingsTest.php [hunk 23]
```diff
@@ -104,8 +109,8 @@ public function testEntityVersionSettingsForm() {
     ];
     foreach ($select_boxes as $index => $select_box) {
       $this->assertFalse($selects[$index]->isVisible());
-      $this->assertEqual($selects[$index]->getAttribute('name'), $select_box['name']);
-      $this->assertEqual($selects[$index]->getAttribute('disabled'), $select_box['disabled']);
+      $this->assertEquals($selects[$index]->getAttribute('name'), $select_box['name']);
+      $this->assertEquals($selects[$index]->getAttribute('disabled'), $select_box['disabled']);
```
Comment (claudiu-cristea) [inline tests/src/FunctionalJavascript/EntityVersionSettingsTest.php:113]: Same here, the expectation should be the 1st param.

File: modules/entity_version_workflows/tests/src/Kernel/EntityVersionWorkflowsTest.php [hunk 24]
```diff
@@ -79,12 +72,12 @@ public function testEntityVersionWorkflow() {
...
     ];
-    /** @var \Drupal\node\NodeInterface $node */
-    $node = $this->nodeStorage->create($values);
+
+    $node = Node::create($values);
     $node->save();
 
     // There is no default value so all versions should be 0.
-    $this->assertEqual('draft', $node->moderation_state->value);
+    $this->assertEquals('draft', $node->moderation_state->value);
```
Comment (claudiu-cristea) [inline modules/entity_version_workflows/tests/src/Kernel/EntityVersionWorkflowsTest.php:80]: This can be `::assertSame()` (which stand for === comparison, while assertEquals is ==).

File: composer.json [hunk 25]
```diff
@@ -6,26 +6,20 @@
     "minimum-stability": "dev",
     "prefer-stable": true,
     "require": {
-        "drupal/core": "^8.8",
-        "php": ">=7.2"
+        "php": ">=7.3",
+        "drupal/core": "^8.9 || ^9.1"
```
Comment (claudiu-cristea) [inline composer.json:10]: We cannot allow a version lower than 8.9.17 as <= 8.9.17 had a vulnerability. See https://www.drupal.org/sa-core-2021-004

**Review by claudiu-cristea — COMMENTED** [review]
I'm requesting changes.
